app/namespace.go

```diff
@@ -978,6 +1057,206 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
+					Usage:   "Get export sink",
+					Flags: []cli.Flag{
+						NamespaceFlag,
+						sinkNameFlag,
+					},
+					Action: func(ctx *cli.Context) error {
+						request := &namespaceservice.GetExportSinkRequest{
+							Namespace: ctx.String(NamespaceFlagName),
+							SinkName:  ctx.String(sinkNameFlag.Name),
+						}
+
+						resp, err := c.client.GetExportSink(c.ctx, request)
+						if err != nil {
+							return err
+						}
+
+						return PrintProto(resp.Sink)
+					},
+				},
+				{
+					Name:    "delete",
+					Aliases: []string{"d"},
+					Usage:   "Delete export sink",
+					Flags: []cli.Flag{
+						NamespaceFlag,
+						sinkNameFlag,
+						ResourceVersionFlag,
+						RequestIDFlag,
+					},
+					Action: func(ctx *cli.Context) error {
+						nameSpaceName := ctx.String(NamespaceFlagName)
+						sinkName := ctx.String(sinkNameFlag.Name)
+						resourceVersion, err := c.getExportSinkResourceVersion(ctx, nameSpaceName, sinkName)
+						if err != nil {
+							return err
+						}
+
+						deleteRequest := &namespaceservice.DeleteExportSinkRequest{
+							Namespace:       nameSpaceName,
+							SinkName:        sinkName,
+							ResourceVersion: resourceVersion,
+							RequestId:       ctx.String(RequestIDFlagName),
+						}
+
+						detelteResp, err := c.client.DeleteExportSink(c.ctx, deleteRequest)
+						if err != nil {
+							return err
+						}
+
+						return PrintProto(detelteResp.RequestStatus)
+					},
+				},
+				{
+					Name:    "list",
+					Aliases: []string{"l"},
+					Usage:   "List export sinks",
+					Flags: []cli.Flag{
+						NamespaceFlag,
+						pageSizeFlag,
+						pageTokenFlag,
+					},
+					Action: func(ctx *cli.Context) error {
+						request := &namespaceservice.ListExportSinksRequest{
+							Namespace: ctx.String(NamespaceFlagName),
+							PageSize:  int32(pageSizeFlag.Value),
+							PageToken: ctx.String(pageTokenFlag.Name),
+						}
+
+						resp, err := c.client.ListExportSinks(c.ctx, request)
+						if err != nil {
+							return err
+						}
+
+						return PrintProto(resp)
+					},
+				},
+				{
+					Name:    "update",
+					Aliases: []string{"u"},
+					Usage:   "Update export sink",
+					Flags: []cli.Flag{
+						NamespaceFlag,
+						sinkEnabledFlag,
+						sinkAssumedRoleFlagOptional,
+						s3BucketFlagOptional,
+						ResourceVersionFlag,
+						kmsArnFlag,
+						RequestIDFlag,
+					},
+					Action: func(ctx *cli.Context) error {
+						nameSpaceName := ctx.String(NamespaceFlagName)
+						sinkName := ctx.String(sinkNameFlag.Name)
+						sink, err := c.getExistingExportSink(ctx, nameSpaceName, sinkName)
+						if err != nil {
+							return err
+						}
+
+						resourceVersion, err := c.getExportSinkResourceVersionWithSink(ctx, sink, nameSpaceName, sinkName)
+						if err != nil {
+							return err
+						}
+
+						if !ctx.IsSet(sinkEnabledFlag.Name) && !ctx.IsSet(sinkAssumedRoleFlagOptional.Name) && !ctx.IsSet(s3BucketFlagOptional.Name) && !ctx.IsSet(kmsArnFlag.Name) {
```

Comment:
what does it mean to update the kmsArnFlag? We should probably update the description to indicate that this is ONLY used for template generation?

Also what does the experience look like if customer tries to write to a s3 bucket where the role doesn't have appropriate KMS permissions? is that status/error message returned useful? we might want to test this scenario